integration/single/test_function_with_http_api_and_auth.py

```diff
@@ -1,11 +1,27 @@
+import logging
+from unittest.case import skipIf
+
+from tenacity import stop_after_attempt, retry_if_exception_type, after_log, wait_exponential, retry, wait_random
+
 from integration.helpers.base_test import BaseTest
+from integration.helpers.resource import current_region_does_not_support
+
+LOG = logging.getLogger(__name__)
 
 
+@skipIf(current_region_does_not_support(["HttpApi"]), "HttpApi is not supported in this testing region")
 class TestFunctionWithHttpApiAndAuth(BaseTest):
     """
     AWS::Lambda::Function tests with http api events and auth
     """
 
+    @retry(
```

Comment:
Could there be issues surrounding the fact that the stacks were not cleaned up correctly, is that a possibility here?

Reply:
Nope for each stack the stack name is generated with a unique hash, random.choice(string.ascii_lowercase) for i in range(RANDOM_SUFFIX_LENGTH), so a failed clean up will never affect stack creation